src/utils.ts

```diff
@@ -124,24 +129,32 @@ export const recursivelyDownloadTx = async (
 
   if (parsedTx.parents.length > 2) {
     // We downloaded a block, we should ignore it
-    return recursivelyDownloadTx(blockId, txIds, data);
+    return recursivelyDownloadTx(blockId, blockHeight, txIds, data);
   }
 
-  // If the first_block from the downloaded tx is not from the block we are searching, we should ignore it.
-  if (meta.first_block !== blockId) {
-    return recursivelyDownloadTx(blockId, txIds, data);
+  // Transaction was already confirmed by a different block in the past
+  if (meta.first_block && meta.first_block !== blockId) {
+    const firstBlockResponse = await downloadTx(meta.first_block);
+
+    // If the transaction was confirmed by an older block, ignore it as it was
+    // already sent to the wallet-service
+    if (firstBlockResponse.tx.height < blockHeight) {
```

Comment:
Should we add an assert when `firstBlockResponse.tx.height === blockHeight`?

Reply:
Why? Is there any chance the fullnode would respond a different block here?